RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate NotificationPreferencesView to Typescript

## Proposed changes
The proposal of this PR is migrate `NotificationPreferencesView` to TypeScript.

## Issue(s)	
N/A

## How to test or reproduce
On the app, open a chat, go to actions view -> notifications, then check if everything still works normally.

## Screenshots
N/A

## Types of changes




## Checklist



## Further comments
N/A
